lib/heimdall_tools/aws_config_mapper.rb

```diff
@@ -0,0 +1,292 @@
+
+NOT_APPLICABLE_MSG = 'No AWS resources found to evaluate complaince for this rule'.freeze
+INSUFFICIENT_DATA_MSG = 'Not enough data has been collectd to determine compliance yet.'.freeze
+
+##
+# HDF mapper for use with AWS Config rules.
+#
+# Ruby AWS Ruby SDK for ConfigService: 
+# - https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/ConfigService/Client.html
+#
+# rubocop:disable Metrics/AbcSize, Metrics/ClassLength
+module HeimdallTools
+  class AwsConfigMapper
+    def initialize(addl_mapping, verbose = false)
+      @verbose = verbose
+      @default_mapping = get_rule_mapping(AWS_CONFIG_MAPPING_FILE)
+      @addl_mapping = addl_mapping.nil? ? {} : get_rule_mapping(addl_mapping)
+      @client = Aws::ConfigService::Client.new
+      @issues = get_all_config_rules
+    end
+
+    ##
+    # Convert to HDF
+    #
+    # If there is overlap in rule names from @default_mapping and @addl_mapping, 
+    # then the rule will be listed twice in the HDF output.
+    # 
+    # All rules that exist in AWS Config and in @addl_mapping will have the rule
+    # name listed in HDF as "USR DEF <rule name>".
+    def to_hdf
+      controls = @issues.filter_map do |issue|
+        @item = {}
+        @item['id']              = issue[:config_rule_name]
+        @item['title']           = issue[:config_rule_name]
+        @item['desc']            = issue[:description]
+        @item['impact']          = 0.5
+        @item['tags']            = hdf_tags(issue, @default_mapping)
+        @item['descriptions']    = hdf_descriptions(issue)
+        @item['refs']            = NA_ARRAY
+        @item['source_location'] = { ref: issue[:config_rule_arn], line: 1 }
+        @item['code']            = ''
+        @item['results']         = issue[:results]
+        # Avoid duplicating rules that exist in the custom mapping as 'unmapped' in this loop
+        if @addl_mapping.include?(issue[:config_rule_name]) && !@default_mapping.include?(issue[:config_rule_name])
+          nil
+        else
+          @item
+        end
+      end
+
+      controls += @issues.filter_map do |issue|
+        if @addl_mapping.include? issue[:config_rule_name]
+          @item = {}
+          @item['id']              = "USR DEF #{issue[:config_rule_name]}"
+          @item['title']           = "USR DEF #{issue[:config_rule_name]}"
+          @item['desc']            = issue[:description]
+          @item['impact']          = 0.5
+          @item['tags']            = hdf_tags(issue, @addl_mapping)
+          @item['descriptions']    = hdf_descriptions(issue)
+          @item['refs']            = NA_ARRAY
+          @item['source_location'] = { ref: issue[:config_rule_arn], line: 1 }
+          @item['code']            = ''
+          @item['results']         = issue[:results]
+          @item
+        end
+      end
+
+      results = HeimdallDataFormat.new(profile_name: 'AWS Config',
+                                       version: '0.0',
```

Comment:
Right now I have this set to `0.0`. Should this be left alone or set to something else like the AWS Config SDK version?

Reply:
Gem version can be accessed with `Aws::ConfigService::GEM_VERSION`